app/src/main/java/org/oppia/app/options/OptionsFragment.kt

```diff
@@ -0,0 +1,185 @@
+
+    bindPreferenceSummaryToValueListener.onPreferenceChange(
+      preference, PreferenceManager
+        .getDefaultSharedPreferences(preference.context)
+        .getString(preference.key, typeOfValue)
+    )
+  }
+
+  private val bindPreferenceSummaryToValueListener =
+    Preference.OnPreferenceChangeListener { preference, newValue ->
+      val stringValue = newValue.toString()
+      when {
+        preference.key == getString(R.string.key_story_text_size) -> // update the changed Text size  to summary filed
+        {
+          preference.summary = stringValue
+          when (stringValue) {
+            "Small" -> {
+              profileManagementController.updateStoryTextSize(profileId, 16f)
+            }
+            "Medium" -> {
+              profileManagementController.updateStoryTextSize(profileId, 18f)
+            }
+            "Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 20f)
+            }
+            "Extra Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 22f)
+            }
+          }
+        }
+        preference.key == getString(R.string.key_app_language) -> // update the changed language  to summary filed
+        {
+          preference.summary = stringValue
+          profileManagementController.updateAppLanguage(profileId, stringValue)
+        }
+        preference.key == getString(R.string.key_default_audio) -> // update the changed audio language  to summary filed
+        {
+          preference.summary = stringValue
+          profileManagementController.updateAudioLanguage(profileId, stringValue)
+        }
+      }
+      true
+    }
+
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    // Check which request we're responding to
+    when (requestCode) {
+      1 -> {
+        val textSize = data!!.getStringExtra("MESSAGE") as String
+        bindPreferenceSummaryToValue(textSize, findPreference(getString(R.string.key_story_text_size)))
+      }
+      2 -> {
+        val appLanguage = data!!.getStringExtra("MESSAGE") as String
+        bindPreferenceSummaryToValue(appLanguage, findPreference(getString(R.string.key_app_language)))
+      }
+      else -> {
+        val audioLanguage = data!!.getStringExtra("MESSAGE") as String
+        bindPreferenceSummaryToValue(audioLanguage, findPreference(getString(R.string.key_default_audio)))
+      }
+    }
+
+  }
+
+  private fun getProfileData(): LiveData<Profile> {
+    return Transformations.map(profileManagementController.getProfile(profileId), ::processGetProfileResult)
+  }
+
+  private fun subscribeToProfileLiveData() {
+    getProfileData().observe(activity, Observer<Profile> {
+      storyTextSize = it.storyTextSize
+      appLanguage = it.appLanguage
+      audioLanguage = it.audioLanguage
+      logger.e("OptionsFragment", "Failed to retrieve profile"+ storyTextSize)
```

Comment:
Update: `logger.e("OptionsFragment", "Failed to retrieve profile$storyTextSize")`

Reply:
Removed this log as it was added for debugging.